x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/analytics/action/AnalyticsStatsAction.java

```diff
@@ -96,110 +107,52 @@ public Response(ClusterName clusterName, List<NodeResponse> nodes, List<FailedNo
             out.writeList(nodes);
         }
-
-        @Override
-        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startArray("stats");
-            for (NodeResponse node : getNodes()) {
-                node.toXContent(builder, params);
-            }
-            builder.endArray();
-
-            return builder;
-        }
     }
 
-    public static class NodeResponse extends BaseNodeResponse implements ToXContentObject {
-        static final ParseField BOXPLOT_USAGE = new ParseField("boxplot_usage");
-        static final ParseField CUMULATIVE_CARDINALITY_USAGE = new ParseField("cumulative_cardinality_usage");
-        static final ParseField STRING_STATS_USAGE = new ParseField("string_stats_usage");
-        static final ParseField TOP_METRICS_USAGE = new ParseField("top_metrics_usage");
-        static final ParseField T_TEST_USAGE = new ParseField("t_test_usage");
-
-        private final long boxplotUsage;
-        private final long cumulativeCardinalityUsage;
-        private final long stringStatsUsage;
-        private final long topMetricsUsage;
-        private final long ttestUsage;
-
-        public NodeResponse(DiscoveryNode node, long boxplotUsage, long cumulativeCardinalityUsage, long stringStatsUsage,
-                long topMetricsUsage, long ttestUsage) {
+    public static class NodeResponse extends BaseNodeResponse {
+        private final EnumCounters<Item> counters;
+
+        public NodeResponse(DiscoveryNode node, EnumCounters<Item> counters) {
             super(node);
-            this.boxplotUsage = boxplotUsage;
-            this.cumulativeCardinalityUsage = cumulativeCardinalityUsage;
-            this.stringStatsUsage = stringStatsUsage;
-            this.topMetricsUsage = topMetricsUsage;
-            this.ttestUsage = ttestUsage;
+            this.counters = counters;
         }
 
         public NodeResponse(StreamInput in) throws IOException {
             super(in);
-            if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
-                boxplotUsage = in.readVLong();
-            } else {
-                boxplotUsage = 0;
-            }
-            cumulativeCardinalityUsage = in.readZLong();
-            if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
-                stringStatsUsage = in.readVLong();
-                topMetricsUsage = in.readVLong();
-            } else {
-                stringStatsUsage = 0;
-                topMetricsUsage = 0;
-            }
             if (in.getVersion().onOrAfter(Version.V_7_8_0)) {
-                ttestUsage = in.readVLong();
+                counters = new EnumCounters<>(in, Item.class);
             } else {
-                ttestUsage = 0;
+                counters = new EnumCounters<>(Item.class);
+                if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
+                    counters.inc(Item.BOXPLOT, in.readLong());
```

Comment:
I think this should be a `readVLong()` instead of `readLong()`?

Reply:
Good catch!